pkg/envoy/registry/registry.go

```diff
@@ -15,22 +13,40 @@ func NewProxyRegistry(mapper ProxyServiceMapper, msgBroker *messaging.Broker) *P
 	}
 }
 
-// RegisterProxy implements MeshCatalog and registers a newly connected proxy.
+// RegisterProxy registers a newly connected proxy.
 func (pr *ProxyRegistry) RegisterProxy(proxy *envoy.Proxy) {
-	pr.connectedProxies.Store(proxy.UUID.String(), connectedProxy{
-		proxy:       proxy,
-		connectedAt: time.Now(),
-	})
+	pr.connectedProxies.Store(proxy.UUID.String(), proxy)
 	log.Debug().Str("proxy", proxy.String()).Msg("Registered new proxy")
 }
 
+// GetConnectedProxy loads a connected proxy from the registry.
+func (pr *ProxyRegistry) GetConnectedProxy(uuid string) (*envoy.Proxy, bool) {
+	p, ok := pr.connectedProxies.Load(uuid)
+	if !ok {
+		return nil, false
+	}
+	return p.(*envoy.Proxy), true
+}
+
 // UnregisterProxy unregisters the given proxy from the catalog.
 func (pr *ProxyRegistry) UnregisterProxy(p *envoy.Proxy) {
 	pr.connectedProxies.Delete(p.UUID.String())
 	log.Debug().Msgf("Unregistered proxy %s", p.String())
 }
 
 // GetConnectedProxyCount counts the number of connected proxies
+// TODO(steeling): switch to a regular map with mutex so we can get the count without iterating the entire list.
```

Comment:
Can you create an issue to track this future work and add it to vFuture?

Reply:
done! and squashed to 1 commit